Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=bb875b38dc5e343bdb696b2eab8233e4d195e208
Commit:     bb875b38dc5e343bdb696b2eab8233e4d195e208
Parent:     f3b8436ad9a8ad36b3c9fa1fe030c7f38e5d3d0b
Author:     Dan Carpenter <error27@gmail.com>
AuthorDate: Mon Jan 26 15:00:58 2009 +0100
Committer:  Miklos Szeredi <mszeredi@suse.de>
CommitDate: Mon Jan 26 15:00:58 2009 +0100

    fuse: fix NULL deref in fuse_file_alloc()
    
    ff is set to NULL and then dereferenced on line 65.  Compile tested only.
    
    Signed-off-by: Dan Carpenter <error27@gmail.com>
    Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
    CC: stable@kernel.org

 fs/fuse/file.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/fs/fuse/file.c b/fs/fuse/file.c
index e816264..d9fdb7c 100644
--- a/fs/fuse/file.c
+++ b/fs/fuse/file.c
@@ -54,7 +54,7 @@ struct fuse_file *fuse_file_alloc(struct fuse_conn *fc)
 		ff->reserved_req = fuse_request_alloc();
 		if (!ff->reserved_req) {
 			kfree(ff);
-			ff = NULL;
+			return NULL;
 		} else {
 			INIT_LIST_HEAD(&ff->write_entry);
 			atomic_set(&ff->count, 0);
